### PR TITLE
Measure owned element by request

### PR DESCRIPTION
**Pull request by zhouyx**
For #8768 
@dvoytenko `mutateElement` itself doesn't work, because we won't do measure if element has owner. [here](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1292).

### CODE REVIEW

File: extensions/amp-sticky-ad/1.0/amp-sticky-ad.js [hunk 1]
```diff
@@ -58,15 +58,22 @@ class AmpStickyAd extends AMP.BaseElement {
...
         'amp-sticky-ad must have a single amp-ad child');
 
     this.ad_ = children[0];
     this.setAsOwner(this.ad_);
+
+    this.ad_.whenBuilt().then(() => {
+      this.mutateElement(() => {
+        toggle(this.element, true);
+      }).then(() => {
+        this.scheduleMeasure(this.ad_);
```
Comment (zhouyx) [inline extensions/amp-sticky-ad/1.0/amp-sticky-ad.js]: need vsync I assume

**Comment by dvoytenko** [comment]
@zhouyx Hmm. I wonder if maybe we can allow owned elements to proceed with re-measurement when it was requested directly?

**Comment by zhouyx** [comment]
How would we define a remeasure is requested directly? Is a remeasure requested from its parent regarded directly?
Then what about making [`requestMeasure`](https://github.com/ampproject/amphtml/blob/master/src/service/resource.js#L483) taking an optional param `requestByParent`, and only set `isMeasureRequested_` value to true if an element is owned by parent and measure request is from its parent. Then I can remove the check of `hasOwner()`.

Owner would then really have to manage its owned children well. (though it is already the case now)

**Comment by dvoytenko** [comment]
@zhouyx I meant that `mutateElement` calls `requestRemeasure` on affected AMP elements so maybe we should allow the remeasure to actually happen [here](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1292).

**Comment by zhouyx** [comment]
I believe [this](https://github.com/ampproject/amphtml/pull/8815/files#diff-80b9cea934a00a64f15d14cb656dfad4R1292) is what you want. I checked `requestMeasure()` is called when a resource is first added, and in `mutateElement` I think both should be valid case to measure an owned element.

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -1289,7 +1289,8 @@ export class Resources {
             relayoutAll || relayoutTop != -1) {
       for (let i = 0; i < this.resources_.length; i++) {
         const r = this.resources_[i];
-        if (r.hasOwner()) {
+        if (r.hasOwner() && !r.isMeasureRequested()) {
```
Comment (jridgewell) [inline src/service/resources-impl.js]: 👍

Comment (dvoytenko) [inline src/service/resources-impl.js]: Yes.

File: extensions/amp-sticky-ad/1.0/amp-sticky-ad.js [hunk 3]
```diff
@@ -58,15 +58,20 @@ class AmpStickyAd extends AMP.BaseElement {
...
-    toggle(this.element, true);
     this.element.classList.add('i-amphtml-sticky-ad-layout');
     const children = this.getRealChildren();
     user().assert((children.length == 1 && children[0].tagName == 'AMP-AD'),
         'amp-sticky-ad must have a single amp-ad child');
 
     this.ad_ = children[0];
     this.setAsOwner(this.ad_);
+
+    this.ad_.whenBuilt().then(() => {
```
Comment (dvoytenko) [inline extensions/amp-sticky-ad/1.0/amp-sticky-ad.js]: Will we do the same in sticky 0.1?

**Comment by dvoytenko** [comment]
This is LGTM on my side. Will we do the same for 0.1?

File: extensions/amp-sticky-ad/1.0/amp-sticky-ad.js (hunk 3 above)
Comment (zhouyx) [inline extensions/amp-sticky-ad/1.0/amp-sticky-ad.js]: I have a sprint task to deprecate the code of sticky 0.1. So I think it's ok to only fix this in 1.0.

**Pull request merged**